### ISSUE TITLE
MDAnalysis.transformations not automatically imported

### ISSUE THREAD

**Issue by woodg07** [MDAnalysis/mdanalysis#2246]
The documentation on the transformations is not clear:

I tried all of these:
```python
MDAnalysis.transformations.center(ag,center='mass')(ts)
MDAnalysis.lib.transformations.center(ag,center='mass')(ts)
MDAnalysis.lib.transformations.center_in_box(ag,center='mass')(ts)
MDAnalysis.transformations.center_in_box(ag,center='mass')(ts)
```
and they all throw:
```
AttributeError: module 'MDAnalysis' has no attribute ....?
```
and then got crazy and tried
```python
MDAnalysis.translate.center_in_box(ag,center='mass')(ts)
MDAnalysis.translate.center(ag,center='mass')(ts)
```

**Comment by richardjgowers** [comment]
Hi @woodg07 

Yep the docs aren't the best sometimes.  I think it looks like you're trying to use `center_in_box` which will put the center of mass into the center of the unit cell each timestep.  I think this should work:

```python
import MDAnalysis as mda
from MDAnalysis import transformations

u = mda.Universe(..., ...)

ag = u.atoms

u.trajectory.add_transformations(transformations.center_in_box(ag, center='mass'))

```

Hope that helps!

**Comment by zemanj** [comment]
@woodg07 
As far as I can tell, there's nothing really wrong with your attempts. The error message
```Python
AttributeError: module 'MDAnalysis' has no attribute 'transformations'
```
is because MDAnalysis doesn't import its `transformations` module by default (@richardjgowers why is that?). So, the line in @richardjgowers' code snippet which fixes the error is this one:
```Python
from MDAnalysis import transformations
```
Please note that the different transformations provided by this module are not ordinary functions, but rather function factories. This means that calling `transformations.center_in_box(ag,center='mass')` won't execute the transformation, but instead create and return a *function object* capable of performing the desired transformation. It might be a bit clearer if it is used like this:
```Python
from MDAnalysis import transformations
#create transformation function:
my_transformation_function = transformations.center_in_box(ag, center='mass')
#use transformation in every time step:
for ts in u.trajectory:
    my_transformation_function(ts)
```
Here, I used the `for` loop over the trajectory only for demonstration purposes. The more appropriate way to use transformations is the one @richardjgowers has shown, i.e., adding the transformation to the trajectory so that it is automatically applied whenever the time step of the trajectory changes.

**Comment by woodg07** [comment]
Thanks @richardjgowers 

I appreciate your response and will give this a go.  BTW I really enjoy this package it is very helpful keep up the good work

**Comment by richardjgowers** [comment]
No worries, glad it’s useful for you! We’ve also got help.mdanalysis.org in the future.

@zemanj not sure - I’m guessing that transformations is optional so it can have exotic imports?

**Comment by orbeckst** [comment]
@richardjgowers @zemanj @jbarnoud I don't think that there's a good reason for _not_ importing it on package load. Probably we didn't auto-import because it's still kind-of in development but it is "in the core" (i.e., neither analysis nor visualization). It should also not have any "exotic imports" – according to [Style Guide: Imports](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide#importing-modules) it needs to import all modules at the top and it can only use the [core module dependencies](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide#list-of-core-module-dependencies).

tldr: I will open an issue to add the import!

**Comment by orbeckst** [comment]
Actually – reopen this one and rename.

**Comment by orbeckst** [comment]
Btw, this issue also lets the binder notebook https://hub.mybinder.org/user/mdanalysis-binder-notebook-ezd1edta/notebooks/notebooks/transformations/on-the-fly-transformations.ipynb fail.

**Comment by orbeckst** [comment]
We can't import `transformations` at the top level because inside it imports stuff from `analysis`. 

I can only fix the docs.

**Comment by richardjgowers** [comment]
@orbeckst if we pushed the `.analysis` imports to inside the function calls, then we could make `.transformation` import at package import.  Would just then need to make sure errors are sensible & early enough (you tried to create transformation_X, this needs networkx or something)

**Comment by orbeckst** [comment]
I think we decided that we do not like imports inside functions, see [Style Guide: General rules for importing](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide#general-rules-for-importing).

We could treat `MDAnalysis.transformations` as akin to `MDAnalysis.analysis`. We should then define the rules, though. For right now, I went with conservative & save.
